### PR TITLE
added test to cover edge case pk variations

### PR DESCRIPTION
**Pull request by manand31**
# Description of change
Added test to cover edge case for pk variations

# QA steps
 - execute the test - test_mongodb_id_pk_variations.py
 
# Risks

# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/test_mongodb_id_pk_variations.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+def generate_docs_string_id():
+    docs = []
+    docs.append({"_id": 'primary_key', "string_field": random_string_generator()})
+    docs.append({"_id": 'secondary_key', "string_field": random_string_generator()})
+    return docs
+
+
+def generate_docs_binary_id():
+    docs = []
+    docs.append({"_id": '10101011', "string_field": random_string_generator()})
```
Comment (kspeer825) [inline tests/test_mongodb_id_pk_variations.py:51]: I think this may just be a string of 1's and 0's and not actually a binary data type. For python you can represent a binary value as an integer  of 1's and 0's prefixed by `0b` so  instead of `'10101011'` it would be `0b10101011`. There are other ways to represent this like using `bytes` and `bytearray` functions. I would check out the pymongo documentations to verify each of these datatypes will be encoded as you expect them to. Here is one page I found for using bson (binary json) https://pymongo.readthedocs.io/en/stable/api/bson/index.html.

File: tests/test_mongodb_id_pk_variations.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        # streams that we synced are the ones that we expect to see
+        record_count_by_stream = runner.examine_target_output_file(self,
+                                                                   conn_id,
+                                                                   self.expected_sync_streams(),
+                                                                   self.expected_pks())
+
+        records_by_stream = runner.get_records_from_target_output()
+
+        # verify if we are capturing all the data for all the streams
+        self.assertEqual(record_count_by_stream, self.expected_record_count())
```
Comment (kspeer825) [inline tests/test_mongodb_id_pk_variations.py:228]: I think there should be more to this test. 

- Validate the values and datatypes of the primary key rows in the target.
- Validate that the _id field is properly marked in metadata as the `table-key-property`.
- Do this test for multiple replication methods, depending on the tap implementation these datatypes may be decoded differently depending on the method.

**Review by kspeer825 — CHANGES_REQUESTED** [review]

File: tests/test_mongodb_id_pk_variations.py (hunk 1 above)
Comment (manand31) [reply tests/test_mongodb_id_pk_variations.py:51]: done

File: tests/test_mongodb_id_pk_variations.py (hunk 2 above)
Comment (manand31) [reply tests/test_mongodb_id_pk_variations.py:274]: done

File: tests/test_mongodb_id_pk_variations.py [hunk 3]
```diff
@@ -108,7 +113,8 @@ def setUp(self):
             client["simple_db"]["coll_with_date_id"].insert_many(generate_docs_date_id())
             client["simple_db"]["coll_with_32_bit_int_id"].insert_many(generate_docs_32_bit_int_id())
             client["simple_db"]["coll_with_64_bit_int_id"].insert_many(generate_docs_64_bit_int_id())
-            client["simple_db"]["coll_with_128_decimal_id"].insert_many(generate_docs_128_decimal_id())
+            # cannot have decimal128 is not a supported replication key type
+            #client["simple_db"]["coll_with_128_decimal_id"].insert_many(generate_docs_128_decimal_id())
```
Comment (kspeer825) [inline tests/test_mongodb_id_pk_variations.py:117]: Not clear on why this is removed. 
1. Does the primary key need to also be the replication key for this test, shouldn't the focus be on the primary key?
2. Where does the 'not supported' limitation come from? Is it the tap, the source db, the client used in the test?
3. If the datatype does not apply to this scenario, can it just be removed from the test entirely?

Comment (manand31) [reply tests/test_mongodb_id_pk_variations.py:117]: The limitation is from the tap, when we select _id to be decimal128 the tap throws an error that it is not a supported replication key type. So this datatype does not apply to this scenario.

Comment (manand31) [reply tests/test_mongodb_id_pk_variations.py:117]: yeah so i am removing it from the test

**Review by kspeer825 — APPROVED** [review]
